monitor_correct_errors/models/monitor_correct_errors.py

```diff
@@ -0,0 +1,336 @@
+            error_log['action_name'] = "hold"
+
+        except:
+            # Create a reservation if hold has not been placed.
+            self.env.invalidate_all()
+            self._create_reservation(child, error_log)
+
+    def _create_reservation(self, child, error_log):
+        '''
+        Create a new reservation.
+        :param child: the child to reserve
+        :return: -
+        '''
+
+        # Check if there is already a reservation for that child.
+        previous_reservation = self.env['compassion.reservation'].search([
+            ('child_global_id', '=', child.global_id),
+            ('state', '!=', 'expired')])
+
+        # If not, we create a new reservation
+        if not previous_reservation:
+            reservation = self.env['compassion.reservation'].create({
+                'child_id': child.id,
+                'reservation_type': 'child',
+                'reservation_expiration_date': self.env[
+                    'compassion.hold'].get_default_hold_expiration(
+                    HoldType.CONSIGNMENT_HOLD),
+                'expiration_date': self.env[
+                    'compassion.hold'].get_default_hold_expiration(
+                    HoldType.CONSIGNMENT_HOLD),
+                'comments': "Automatic reservation place for missing hold"
+            })
+            # Commit when we create a reservation to avoid its deletion
+            self.env.cr.commit()    # pylint: ignore=invalid-commit
+            # Activation of the reservation
+            reservation.handle_reservation()
+
+            error_log['action'] = "new reservation created"
+            error_log['action_model'] = "compassion.reservation"
+            error_log['action_id'] = reservation.id
+            error_log['action_name'] = "reservation"
+
+        else:
+            error_log['action'] = "reservation already exist"
+            error_log['action_model'] = "compassion.reservation"
+            error_log['action_id'] = previous_reservation.id
+            error_log['action_name'] = "reservation"
+
+    def _sent_mail(self):
+
+        message_body = ""
+        error_type_list = {}
+        for error_log in self.error_log_list:
+            if error_log.error_type not in error_type_list:
+                error_type_list[error_log.error_type] = []
+
+            error_type_list[error_log.error_type].append(error_log)
+
+        for error_type in error_type_list:
+
+            # Get the error type from the list.
+            message_body += "<h4>" + error_type + "</h4>"
+            # Get the error type from the list.
+            error_list = error_type_list[error_type]
+
+            message_body += "<table class=\"editorDemoTable\" border=\"1\">"
+            message_body += "<tr>"
+            message_body += "<td><h4>" + "Record" + "</h4></td>"
+            message_body += "<td><h4>" + "Error type" + "</h4></td>"
+            message_body += "<td><h4>" + "Action" + "</h4></td>"
+
+            for error in error_list:
```

Comment:
Here you will be able to do: `for error in self.error_ids.filtered(lambda e: e.type == error_type):`